### ISSUE TITLE
Implement r.pathspec(...)

### ISSUE THREAD

**Issue by NotJustAnna** [rethinkdb/rethinkdb-java#16]
### This issue is a continuation of https://github.com/rethinkdb/rethinkdb/issues/5007

Implement the discussed at the issue.

From the original issue:

> Right now pathspecs are kind hairy due to Java's lack of nice hashmap and List literals:
> 
> in javascript:
> 
> ```js
> r.table('foo').filter({x: {y: {z: "bar"}}})
> ```
> 
> turns into the java query:
> 
> ```java
> r.table("foo").filter(r.hashMap("x", r.hashMap("y", r.hashMap("z", "bar"))))
> ```
> 
> Since depth-first is more common in path-specs than breadth first, it might make sense to have a special `r.pathspec` convenience that creates the appropriate hashmap structure. It might look something like:
> 
> ```java
> r.table("foo").filter(r.pathspec("x", "y", "z", "bar"))
> ```
> 
> I haven't really thought about branching pathspecs too much, maybe there's no syntax that would work well. It's something to think about.

**Comment by NotJustAnna** [comment]
Implemented in `develop/experimental` (#7)

**Comment by NotJustAnna** [comment]
# To be documented:

The best way to solve the breadth problem is pathspec with object:

```java
r.table("foo").filter(
    r.object("x1", r.pathspec("y", "z", "bar"), "x2", r.pathspec("y", "z", "bar")
)
```
